google-workspace.sgnl.yaml

```diff
@@ -0,0 +1,277 @@
+defaultSyncFrequency: HOURLY
+defaultSyncMinInterval: 1
+defaultApiCallFrequency: SECONDLY
+defaultApiCallMinInterval: 1
+type: "GoogleWorkspace-1.0.0"
+# Example Config: 
+# {
+#   "apiVersion": "v1",
+#   "domain": "sgnldemos.com",
+#   "filters": {
+#     "user": {
+#       "showDeleted": true,
+#       "query": "isAdmin=false"
+#     },
+#     "group": {
+#       "query": "adminCreated=true"
+#     },
+#     "member": {
+#       "includeDerivedMembership": false,
+#       "roles": "MEMBER"
+#     }
+#   }
+# }
+
+# Adapter Config Field Info:
+# [Required] APIVersion (String): The version of the Google Workspace API to use. Currently, only v1 is supported.
+
+# DISCLAIMER: Exactly one of either 'Customer' or 'Domain' must be set in the adapter config.
+# Read more about the Customer and Domain parameters here: https://developers.google.com/admin-sdk/directory/v1/guides
+
+# [Required/Optional] Customer (String): The unique ID for the customer's Google Workspace account. In case of a multi-domain account,
+# to fetch all entities for a customer, use this field instead of domain.
+
+# [Required/Optional] Domain (String): Use this field to get entities from only one domain.
+
+# [Optional] Filters (Map): These filters can be applied to the endpoints that are used to retrieve each entity. 
+
+# For more information on the User endpoint: https://developers.google.com/admin-sdk/directory/reference/rest/v1/users/list
+# [Optional] Filters.UserFilters (Map): These filters can be applied to the User endpoint.
+# [Optional] Filters.UserFilters.ShowDeleted (Bool): If set to true, retrieves the list of deleted users. (Default: false)
+# [Optional] Filters.UserFilters.Query (String): Query is a string for searching entity fields.
+# For more information on constructing user queries: https://developers.google.com/admin-sdk/directory/v1/guides/search-users
+
+# For more information on the Group endpoint: https://developers.google.com/admin-sdk/directory/reference/rest/v1/groups/list
+# [Optional] Filters.GroupFilters (Map): These filters can be applied to the Group endpoint.
+# [Optional] Filters.GroupFilters.Query (String): Query is a string for searching entity fields.
+# For more information on constructing group queries: https://developers.google.com/admin-sdk/directory/v1/guides/search-groups
+
+# For more information on the member endpoint: https://developers.google.com/admin-sdk/directory/reference/rest/v1/members/list
+# [Optional] Filters.MemberFilters (Map): These filters can be applied to the Member endpoint.
+# [Optional] Filters.MemberFilters.IncludeDerivedMembership (Bool): If set to true, includes lists of indirect memberships. (Default: false)
+# [Optional] Filters.MemberFilters.Roles (String): The roles query parameter allows you to retrieve group members by role.
+# Roles is an ENUM with possible values of ["MEMBER", "MANAGER", "OWNER"]
+
+adapterConfig: ewogICAgImFwaVZlcnNpb24iOiAidjEiLAogICAgImRvbWFpbiI6ICJ7e0lucHV0IFJlcXVpcmVkfX0iCn0=
```

Comment:
How about we add a config that contains almost all fields and it's values set to defaults?
Makes it easier for user to configure and tweak!

Reply:
fixed